### ISSUE TITLE
Unable to pass header to websocket server

### ISSUE THREAD

**Issue by newbaluku** [theturtle32/WebSocket-Node#386]
trying to pass an auth header to server as such (client running under Node):
```
const client =  new WebsocketClient();
client.connect('wss://websocketUrl', null, null, { headers: { 'x-appid': process.env.APPID } });

```
but server returns:
```
Server responded with a non-101 status: 401 Unauthorized
Response Headers Follow:
date: Mon, 06 Jul 2020 11:04:20 GMT
content-type: application/json; charset=UTF-8
content-length: 94
connection: keep-alive
x-amz-apigw-id: PP3APFy4SQ0FnDw=

    at WebSocketClient.failHandshake (/var/task/node_modules/websocket/lib/WebSocketClient.js:339:32)
    at ClientRequest.<anonymous> (/var/task/node_modules/websocket/lib/WebSocketClient.js:278:18)
    at ClientRequest.emit (events.js:310:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:596:27)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:119:17)
    at TLSSocket.socketOnData (_http_client.js:469:22)
    at TLSSocket.emit (events.js:310:20)
    at addChunk (_stream_readable.js:286:12)
    at readableAddChunk (_stream_readable.js:268:9)
    at TLSSocket.Readable.push (_stream_readable.js:209:10)
```
if I disabled authentication on server side, it will be able to connect.
Is there anything wrong with the way I called connect with the header param?

**Comment by ragnarok22** [comment]
Same issue

**Comment by AshwinDeTaeye** [comment]
I have the same issue. I want to send the cookie I received after a login.
```    
const headers = { headers: { Cookie: cookieGot } };
const client = new W3CWebSocket(`ws://${ip}/wsapi`, undefined, undefined, headers);
// Type error for headers
```  
When I install also the @types/websocket I can see the following error:

_Argument of type '{ headers: { Cookie: Cookie; }; }' is not assignable to parameter of type 'OutgoingHttpHeaders'._

When I search for the Type of 'OutgoingHttpHeaders' online I can't find any information on what it should contain.

Also, in the docs it says: client.connect('ws://echo.websocket.org/', null, null, null, requestOptions);
This doesn't work in typescript, should all be undefined instead of null.
If I connect to a ws server without the headers, the connection is made and i can send and receive data. 

The ws connection is being made in the main.ts from the electron application.

    "electron": "^13.1.8",
    "@types/websocket": "^1.0.4",
    "websocket": "^1.0.34"

**Comment by toJuanito** [comment]
What I found was that the library W3CWebSocket in browsers does not work with 'OutGoingHttpHeaders' but everything that I found was posted 2-3 years ago.

The only header that I was able to change was 'sec-websocket-protocol' (Second parameter in the constructor).

Actual version of websocket: 
``
"websocket": "^1.0.34"
``
What I did was to pass everything that I need via url parameters and 'handle it' in the server.
